pkg/maglev/maglev.go

```diff
@@ -28,16 +28,30 @@ const (
 
 	// seed=$(head -c12 /dev/urandom | base64 -w0)
 	DefaultHashSeed = "JLfvgnHc2kaSUFaI"
+
+	// DefaultMaxBackends is the default number of backends per service
+	// supported in Maglev. This value is used in initializing the Maglev
+	// permutations.
+	DefaultMaxBackends = 5000
 )
 
 var (
 	seedMurmur uint32
 
 	SeedJhash0 uint32
 	SeedJhash1 uint32
+
+	maxBackends uint64
+
+	// permutation is the slice containing the Maglev permutation calculations.
+	permutation []uint64
 )
 
-func InitMaglevSeeds(seed string) error {
+// InitMaglevSeeds initializes the Maglev subsystem with the seed, the backend
+// table size (m), and the maximum number of backends. Note that `backends` is
+// not a hard limit, but it is used in preallocating the permutation slice, for
+// performance purposes.
+func InitMaglevSeeds(seed string, m, backends uint64) error {
```

Comment:
I think this function needs to be renamed as it no longer does only the seeds init.